src/azul/service/responseobjects/storage_service.py

```diff
@@ -37,14 +43,144 @@ def put(self, object_key: str, data: bytes, content_type: Optional[str] = None)
 
         return object_key
 
+    @contextmanager
+    def multipart_upload(self, object_key: str, content_type: str):
```

Comment:
I would eliminate this function and turn MultipartUploadHandler into a context manager by adding `__enter__` and `__exit__` methods. It should be cleaner. What smells here is the fact that you have to pass the thread pool and the upload ID to the the MultipartUploadHandler constructor. That indicates that MultipartUploadHandler should manage that state and not this function.

Reply:
I agree. In your opinion, what do you think of turning `MultipartUploadHandler` into a subclass of this class?

Reply:
Dubious. A MultipartUploadHandler is not a StorageService. A StorageService has an MultipartUploadHandler.